src/cryptonote_core/tx_pool.cpp

```diff
@@ -375,16 +375,26 @@ namespace cryptonote
         txpool_tx_meta_t meta;
         if (!m_blockchain.get_txpool_tx_meta(txid, meta))
         {
-          MERROR("Failed to find tx in txpool");
+          MERROR("Failed to find tx_meta in txpool");
           return;
         }
         // don't prune the kept_by_block ones, they're likely added because we're adding a block with those
-        if (meta.kept_by_block)
+        switch (meta.get_relay_method())
         {
-          --it;
-          continue;
+          default:
+            MERROR("Failed to find tx in txpool");
+            return;
+
+          case relay_method::block:
+            --it;
+            continue;
+
+          case relay_method::flood:
+          case relay_method::local:
+          case relay_method::none:
+            break;
```

Comment:
That switch looks a lot more complex that the if that there before, and does not do anything else once the return gets reverted.